### PR TITLE
#2808

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/2808 - [studio-ui] child-content data source should allow user to decide if add or browse options are enabled #2808

### CODE REVIEW

File: static-assets/components/cstudio-forms/data-sources/child-content.js [hunk 1]
```diff
@@ -35,6 +37,16 @@ function(id, form, properties, constraints)  {
 		if(properties[i].name == "type"){
 			this.type = (Array.isArray(properties[i].value))?"":properties[i].value;
 		}
+
+        if(properties[i].name == "enableCreateNew"){
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/child-content.js]: 3ple equals

File: static-assets/components/cstudio-forms/data-sources/child-content.js [hunk 2]
```diff
@@ -35,6 +37,16 @@ function(id, form, properties, constraints)  {
...
+            this.enableCreateNew = properties[i].value == "true" ? true : false;
+            this.defaultEnableCreateNew = false;
+        }
+
+        if(properties[i].name == "enableBrowseExisting"){
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/child-content.js]: ===

File: static-assets/components/cstudio-forms/data-sources/child-content.js [hunk 3]
```diff
@@ -51,111 +63,119 @@ YAHOO.extend(CStudioForms.Datasources.ChildContent, CStudioForms.CStudioFormData
 		
 		var addContainerEl = null;
 
+        var datasourceDef = this.form.definition.datasources,
+            newElTitle = '';
+
+        for(var x = 0; x < datasourceDef.length; x++){
+            if (datasourceDef[x].id == this.id){
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/child-content.js]: ===

File: static-assets/components/cstudio-forms/data-sources/child-content.js [hunk 4]
```diff
@@ -51,111 +63,119 @@ YAHOO.extend(CStudioForms.Datasources.ChildContent, CStudioForms.CStudioFormData
...
-		for(var x = 0; x < datasourceDef.length; x++){
-			if (datasourceDef[x].id == this.id){
-				newElTitle = datasourceDef[x].title;
-			}
-		}
+            var addContainerEl = control.addContainerEl;
+            YAHOO.util.Event.on(createEl, 'click', function () {
+                control.addContainerEl = null;
+                control.containerEl.removeChild(addContainerEl);
+                if (_self.type == "") {
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/child-content.js]: ===

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-forms/data-sources/child-content.js (hunk 1 above)
Comment (vestrada) [reply static-assets/components/cstudio-forms/data-sources/child-content.js]: Done

File: static-assets/components/cstudio-forms/data-sources/child-content.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-forms/data-sources/child-content.js]: Done

File: static-assets/components/cstudio-forms/data-sources/child-content.js (hunk 3 above)
Comment (vestrada) [reply static-assets/components/cstudio-forms/data-sources/child-content.js]: Done

File: static-assets/components/cstudio-forms/data-sources/child-content.js (hunk 4 above)
Comment (vestrada) [reply static-assets/components/cstudio-forms/data-sources/child-content.js]: Done

**Review by rart — APPROVED** [review]

**Pull request merged**